mlflow/utils/_capture_modules.py

```diff
@@ -79,6 +82,12 @@ def _record_imported_module(self, full_module_name):
                     self.imported_modules.add(databricks_module)
                     return
 
+        # special casing for mlflow extras since they may not be required by default
+        if top_level_module == "mlflow":
+            if second_level_module in MLFLOW_MODULES_TO_PACKAGES:
+                self.imported_modules.add(second_level_module)
+                return
```

Comment:
tried to follow the format of the databricks special casing above, but not sure if it's necessary since this really only affects `gateway`